Backend/app.py

```diff
@@ -52,67 +54,99 @@ def getDatabaseClient():
+  flaskResponse =  ResponseCreation.createResponse(res, status)
+  logger.info(str(flaskResponse))
+  return flaskResponse
 
 @app.route('/activate/<activationid>', methods=['GET'])
 def activate(activationid):
+  logger.info(str(request))
   dynamoDBInstance = getDatabaseClient()
   res = user_ctrl.activate(activationid, dynamoDBInstance)
-  if not res.hasErrors():
-    return cr.createResponse(res, 200)
-  else:
-    return cr.createResponse(res, 400)
+  status = 200
+  if res.hasErrors():
+    status = 400
+  flaskResponse = ResponseCreation.createResponse(res, status)
+  logger.info(str(flaskResponse))
+  return flaskResponse
 
 @app.route('/signin/', methods=['POST', 'OPTIONS'])
 @auth.login_required
 def signin():
-  if request.method =='POST':
-    userCredentials = {'email': request.authorization.username, 'password': request.authorization.password}
-    response = ControllerResponse()
-    token = MentiiAuth.generate_auth_token(userCredentials, appSecret)
-    response.addToPayload('token', token)
-    return cr.createResponse(response, 200)
-  else:
-    return cr.createEmptyResponse(200)
+  logger.info(str(request))
+  status = 200
+  if request.method =='OPTIONS':
+    flaskResponse = ResponseCreation.createEmptyResponse(status)
+    logger.info(str(flaskResponse))
+    return flaskResponse
+  userCredentials = {'email': request.authorization.username, 'password': request.authorization.password}
+  response = ControllerResponse()
+  token = MentiiAuth.generate_auth_token(userCredentials, appSecret)
+  response.addToPayload('token', token)
+  flaskResponse = ResponseCreation.createResponse(response, status)
+  logger.info(str(flaskResponse))
+  return flaskResponse
 
 @app.route('/secure/', methods=['POST', 'OPTIONS'])
 @auth.login_required
 def secure():
   '''
   This is an endpoint for testing authentication
   '''
-  if request.method =='POST':
+  logger.info(str(request))
+  status = 200
+  if request.method =='OPTIONS':
+    flaskResponse = ResponseCreation.createEmptyResponse(status)
+    logger.info(str(flaskResponse))
+    return flaskResponse
     # Do ya thang for the endpoint here. call another class/function etc
     # testing function just returns data on the authenticated user
-    response = ControllerResponse()
-    response.addToPayload('user', g.authenticatedUser)
-    return cr.createResponse(response, 200)
-  else:
-    return cr.createEmptyResponse(200)
+  response = ControllerResponse()
+  response.addToPayload('user', g.authenticatedUser)
+  flaskResponse = ResponseCreation.createResponse(response, status)
+  logger.info(str(flaskResponse))
+  return flaskResponse
 
 if __name__ == '__main__':
-  app.run(host='0.0.0.0', debug=False)
+  MentiiLogging.setupLogger()
+  logger = MentiiLogging.getLogger()
+  logger.info('mentii app starting')
+  try:
+    app.run(host='0.0.0.0', debug=False)
+  except Exception as e:
+    logger.exception(e)
+  logger.warning('mentii app down')
```

Comment:
The warning will fire without an exception, whenever the app stops running, so indentation is correct